server/registry/internal/storage/filtering/filtering.go

```diff
@@ -91,5 +94,12 @@ func NewFilter(filter string, fields map[string]FieldType) (Filter, error) {
 		return Filter{}, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	return Filter{program: prg}, nil
+	sqlCondition, err := cel2sql.Convert(ast)
+	if err != nil {
+		return Filter{}, status.Error(codes.InvalidArgument, err.Error())
+	}
+	// TODO: quick hack to adjust field name
+	sqlCondition = strings.ReplaceAll(sqlCondition, "name", "key")
```

Comment:
Wow, I'm amazed that `cel2sql.Convert()` works with so little modification. I was thinking that we would have to own this parsing/conversion. I expect it will at least need to be a grey box to us (if we use it, we'll at least be stepping inside it to understand/debug it)

Reply:
Eh... well, it "works" for simple cases. It's not complete and it's created specifically for Big Query, so most likely we'll have to fork... and even then, as I mentioned, I think we'll have to have a fallback path for some CEL.

Reply:
(I just read your comment about the failing tests, that's in line with this)